src/inference/weaken.rs

```diff
@@ -487,7 +491,7 @@ where
         self.by_id.values().map(|o| o.to_base()).collect_vec()
     }
 
-    pub fn iter(&self) -> Vec<(Arc<QuantifierPrefix>, &O)> {
+    pub fn as_vec(&self) -> Vec<(Arc<QuantifierPrefix>, &O)> {
```

Comment:
Instead of collecting and returning a `Vec`, you can actually return an iterator. The best way to write the signature for that is like the following:
```rust
pub fn as_iter(&self) -> impl Iterator<Item = (Arc<QuantifierPrefix>, &O)> {
    self.by_id.values().map(|body| (self.prefix.clone(), body))
}
```

What this is saying is that the function returns some type that implements this trait (to hide the fact that you're returning a std::iter::Map iterator).